src/FirebaseDemo.tsx

```diff
@@ -0,0 +1,84 @@
+import { useState, useEffect } from 'react';
+import { db } from '../firebase.config';
+import { collection, getDocs, addDoc, updateDoc, deleteDoc, doc, QuerySnapshot } from 'firebase/firestore';
+import { Box, Button, Flex, Heading, Input, Stack, HStack } from '@chakra-ui/react';
+
+export function FirebaseDemo() {
+  const [newName, setNewName] = useState('');
+  const [newMark, setNewMark] = useState(0);
+
+  const [users, setUsers] = useState<Array<{ id: string; name: string; mark: number }>>([]);
+  const usersCollectionRef = collection(db, 'users');
+
+  const createUser = async () => {
+    await addDoc(usersCollectionRef, { name: newName, mark: Number(newMark) });
+    getUsers();
+  };
+
+  const updateUser = async (id: string, mark: number) => {
+    const userDoc = doc(db, 'users', id);
+    const newFields = { mark: mark + 1 };
+    await updateDoc(userDoc, newFields);
+    getUsers();
+  };
+
+  const deleteUser = async (id: string) => {
+    const userDoc = doc(db, 'users', id);
+    await deleteDoc(userDoc);
+    getUsers();
+  };
+
+  const getUsers = async () => {
```

Comment:
What do you think about adding some Firebase hooks library? (like `react-firebase-hooks`, if there isn't a better alternative)

Reply:
Yes, I can do that. Thanks for suggesting!